ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/LedgerDao.scala

```diff
@@ -74,6 +75,14 @@ trait LedgerDao extends AutoCloseable {
       _.getOrElse(sys.error(s"ledger entry not found for offset: $offset")))(DirectExecutionContext)
   }
 
+  /**
+    * Looks up a Contract given a contract key
+    *
+    * @param key the contract key to query
+    * @return the optional AbsoluteContractId
+    */
+  def lookupKey(key: Node.GlobalKey): Future[Option[AbsoluteContractId]]
```

Comment:
Is it ok that the contract key is read in its own SQL transaction? Doesn't it have to be read together with other `lookupKey` or `lookupContract` queries from some consistent snapshot?

Reply:
yes, it is, see comment in https://github.com/digital-asset/daml/blob/e83c21ae11e939abc0c12a08887381b888df376a/ledger/backend-api/src/main/scala/com/digitalasset/ledger/backend/api/v1/SubmissionHandle.scala#L56